src/class-block-converter.php

```diff
@@ -315,21 +320,44 @@ public function remove_image_args( $url ): string {
 		// Split url.
 		$url_parts = wp_parse_url( $url );
 
-		return $url_parts['scheme'] . '://' . $url_parts['host'] . $url_parts['path'];
+		// Negotiate scheme.
+		$scheme = $url_parts['scheme'] ?? '';
+		if ( empty( $scheme ) ) {
+			$home = get_option( 'home' );
```

Comment:
@dlh01 @kevinfodness My understanding of this feature is that images coming from the HTML source will be uploaded to the media library. With this in mind, I think the `schema` from those source images would not necessarily match the one in WordPress `home` URL value. Also, the original intent of this function was to quickly remove query parameters from the URL, but with the changes introduced by @kevinfodness it would try to access database information (the home URL), which might be an issue when handling a lot of images.

Reply:
Yep, I agree, I'll switch this to a filter that defaults to https